### PR TITLE
[WIP] Organizational model obfuscation and permission mechanism

### PR DESCRIPTION
**Pull request by alexatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-WebServices-Community-Edition/master/.github/CONTRIBUTING.md)
- [] I have provided test coverage for my change (where applicable)

### Description
Serverside obfuscation of data based on organizational association of the requester.

### CODE REVIEW

File: CDP4WebServices.API/Services/Authorization/ObfuscationService.cs [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+            var elementUsages = resourceResponse.OfType<ElementUsage>().ToList();
+            var parameters = resourceResponse.OfType<Parameter>().ToList();
+            var parameterOverrides = resourceResponse.OfType<ParameterOverride>().ToList();
+            var parameterSubscriptions = resourceResponse.OfType<ParameterSubscription>().ToList();
+            var parameterValueSets = resourceResponse.OfType<ParameterValueSet>().ToList();
+            var parameterSubscriptionValueSets = resourceResponse.OfType<ParameterSubscriptionValueSet>().ToList();
+            var parameterOverrideValueSets = resourceResponse.OfType<ParameterOverrideValueSet>().ToList();
+            var parameterGroups = resourceResponse.OfType<ParameterGroup>().ToList();
+            var definitions = resourceResponse.OfType<Definition>().ToList();
+            var citations = resourceResponse.OfType<Citation>().ToList();
```
Comment (samatrhea) [inline CDP4WebServices.API/Services/Authorization/ObfuscationService.cs:80]: i have the impression that the `Alias` dto should be here as well

**Review by samatrhea — CHANGES_REQUESTED** [review]
code looks very good and super clean, some remarks:
  - think you missed the `Alias` class (next to hyperlink and and definition
  - Nice testable code, but I don't see a unit test for the obfuscation service

File: CDP4-Server.sln.DotSettings [hunk 2]
```diff
@@ -242,6 +242,32 @@
 	<s:Boolean x:Key="/Default/CodeStyle/CSharpUsing/QualifiedUsingAtNestedScope/@EntryValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/CodeStyle/EditorConfig/EnableStyleCopSupport/@EntryValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/CodeStyle/EditorConfig/ShowEditorConfigStatusBarIndicator/@EntryValue">True</s:Boolean>
+	<s:String x:Key="/Default/CodeStyle/FileHeader/FileHeaderText/@EntryValue">--------------------------------------------------------------------------------------------------------------------&#xD;
+&lt;copyright file="$FILENAME$" company="RHEA System S.A."&gt;&#xD;
+   Copyright (c) 2015-$CURRENT_YEAR$ RHEA System S.A.&#xD;
+&#xD;
+   Author: Sam Gerené, Merlin Bieze, Alex Vorobiev, Naron Phou, Alexander van Delft, Nathanael Smiechowski,&#xD;
+   Ahmed Ahmed&#xD;
```
Comment (lxatrhea) [inline CDP4-Server.sln.DotSettings:250]: Most of us are using his complete name in headers: Ahmed Abulwafa Ahmed

**Review by lxatrhea — APPROVED** [review]
Looks good.
Somewhat related: Are we sure that Poco's created from these DTO's cannot accidentally be written back to the database? A user's access rights might change during the lifetime of a Session (Cache). Or the user's organizational setting might change.

**Comment by alexatrhea** [comment]
> Looks good.
> Somewhat related: Are we sure that Poco's created from these DTO's cannot accidentally be written back to the database? A user's access rights might change during the lifetime of a Session (Cache). Or the user's organizational setting might change.

This is in progress

**Comment by alexatrhea** [comment]
Handed over to @lxatrhea for unit test fixing and finalization

**Pull request merged**